src/legisdata/main.py

```diff
@@ -1,33 +1,93 @@
+import json
+import mimetypes
 import os
+import pickle
 import time
+from datetime import datetime
+from itertools import chain
 from posixpath import basename
 from random import randrange
-from typing import List
+from typing import Any, Iterable, List, NamedTuple, Union
 
 import requests
 import structlog
 import typer
 from huggingface_hub import HfApi
 from parsel import Selector
+from unstructured.documents.elements import Element, Image, ListItem, Table, Title
+from unstructured.partition.pdf import partition_pdf
 
 app = typer.Typer()
 api = HfApi()
 logger = structlog.get_logger()
 
+class Person(NamedTuple):
+    name: str
+    area: str | None = None
+
+
+class ContentElement(NamedTuple):
+    type: str
+    value: str
+    image: None | str
+
+
+class Meta(NamedTuple):
+    source: str
+    year: int
+    session: int
+    dun: str
+    parse_time: str = str(datetime.now())
+
+
+class Inquiry(NamedTuple):
```

Comment:
Just keeping as a note, this is the structure of the resulting JSON file for an inquiry